### ISSUE TITLE
Implement the main directions control

### ISSUE THREAD

**Issue by smellyshovel** [maplibre/maplibre-gl-directions#58]
The main control should provide at least 2 fields: one for origin and one for destination. Most probably they should also support geocoding.

If the `steps` request option is enabled, the control should also display the routing steps and allow to navigate through the waypoints.

*To be continued...*

**Comment by TheFireCircle** [comment]
Is it possible to expose the route steps in the MapLibreGlDirections object to enable users of the library to build their own Steps-Control? Or did I overlook a variable in the MapLibreGlDirections object that I can use?

**Comment by smellyshovel** [comment]
@TheFireCircle yes, sure. Listen to the `fetchroutesend` event, it will cotain the full response from the API.

In fact, if the plugin implemented the control natively, this is exactly how it would be done as well. So it's not even a "hacky" way.

**Comment by TheFireCircle** [comment]
@smellyshovel Thank you for your fast answer!
